c10/test/util/string_util_test.cpp

```diff
@@ -77,4 +77,58 @@ TEST(StringUtilTest, testStrMulti) {
 }
 } // namespace test_str_multi
 
+namespace test_try_to {
```

Comment:
I'm not sure how the C++ tests work. Do you know which shard and when this is executed in CI ?

Reply:
@albanD I don't know how I can find the exact shard, but when I had a bug in unittest, the CI is able to turn red for some shard... (I couldn't find it anymore since it's overwritten). 

Weirdly we only have 4 checks run on this PR now with the new push.

Reply:
Given the msvc/clang comment below, hopefully this runs on all 3 OS.
Also for future tests that might need cuda or rely on specific hardware. We will need to know exactly where it runs to make sure we have the appropriate coverage.